ConsistencyManager/ConsistencyManager.swift

```diff
@@ -699,16 +698,15 @@ open class ConsistencyManager {
      This function uses the map functionality of the models to generate a new model given a list of modelUpdates.
      It returns a new model, a list of changes (ModelUpdates) and a list of any new models which were not contained in the old model.
      */
-    private func updatedModelFromOriginalModel(_ model: ConsistencyManagerModel, updatedModels: [String: [ConsistencyManagerModel]?], context: Any?) -> (model: ConsistencyManagerModel?, updates: ModelUpdates, newModels: [ConsistencyManagerModel]) {
+    private func updatedModelFromOriginalModel(_ model: ConsistencyManagerModel, updatedModels: [String: ModelChange], context: Any?) -> (model: ConsistencyManagerModel?, updates: ModelUpdates, newModels: [ConsistencyManagerModel]) {
         if let id = model.modelIdentifier {
-            if let replacementModel = updatedModels[id] {
+            if let modelChange = updatedModels[id] {
                 // The id matches, so we should replace this model with a different model
-                // Important: replacementModel could actually be nil here. This is because modelUpdates[id] is actually type: ConsistencyManagerModel??.
-                // So, the let statement only unwraps it once. This is a good thing since if it is nil, we want to delete the model.
                 // At the point, we know that this is an id we care about. Let's see if it's an update or a delete.
-                if let replacementModel = replacementModel {
+                switch modelChange {
+                case .updated(let replacementModels):
                     // It's an update. Let's apply the changes.
-                    let mergedReplacementModel = mergedModelFromModel(model, withUpdates: replacementModel)
+                    let mergedReplacementModel = mergedModelFromModel(model, withUpdates: replacementModels)
```

Comment:
`withUpdates` label seems out of date now. Should either match the type (e.g. `withModelChanges`) or just be `with` since it now has a strong type.